### PR TITLE
Backport CMake related changes

### PR DESCRIPTION
**Pull request by dnakamura**
- Don't stage source files
- Enable DDR support

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile xlinuxcm jdk8

File: closed/OpenJ9.gmk [hunk 1]
```diff
@@ -405,42 +408,93 @@ ifneq (,$(OPENJ9_DEVELOPER_DIR))
...
+	  -DJ9VM_OMR_DIR=$(OPENJ9OMR_TOPDIR) \
+	  -DBUILD_ID=$(BUILD_ID) \
+	  -DJAVA_SPEC_VERSION=8 \
+	  -DOPENJ9_BUILD=true \
+	  $(EXTRA_CMAKE_ARGS) \
+	  $(OPENJ9_TOPDIR)
+	touch $(OUTPUT_ROOT)/vm/cmake.stamp
+run-preprocessors-j9: $(OUTPUT_ROOT)/vm/cmake.stamp
+else # OPENJ9_ENABLE_CMAKE
+run-preprocessors-j9: stage-j9
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: Space before `:`.

File: jdk/make/closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -309,28 +309,31 @@ AC_DEFUN_ONCE([OPENJ9_THIRD_PARTY_REQUIREMENTS],
...
-    printf "tar -xzf freemarker-2.3.8.tar.gz\n"
-    printf "\n"
-    printf "Then run configure with '--with-freemarker-jar=<freemarker_jar>'\n"
-    printf "\n"
-
-    AC_MSG_NOTICE([Could not find freemarker.jar])
-    AC_MSG_ERROR([Cannot continue])
-  fi
+  FREEMARKER_JAR=
+  if test "x$with_cmake" == x ; then
```
Comment (keithc-ca) [inline jdk/make/closed/autoconf/custom-hook.m4]: This doesn't seem to work:
```
$ bash configure --with-boot-jdk=/home/keithc/space/bootjdk8 --with-cmake
[...]
The FreeMarker library is required to build the OpenJ9 build tools
and has to be provided during configure process.
[...]
```

Comment (keithc-ca) [reply jdk/make/closed/autoconf/custom-hook.m4]: Adding `--with-freemarker-jar=/home/keithc/space/freemarker.jar` gets further, but still fails:
```
make[3]: *** No rule to make target '/home/keithc/space/jdk08/build/linux-x86_64-normal-server-release/vm/j9ddr.dat', needed by '/home/keithc/space/jdk08/build/linux-x86_64-normal-server-release/jdk/ddr/classes.done'.  Stop.
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [openj9.ddr-gen] Error 2
/home/keithc/space/jdk08/closed/make/Images.gmk:48: recipe for target 'openj9.ddr-gen' failed
```

Comment (dnakamura) [reply jdk/make/closed/autoconf/custom-hook.m4]: Looks like I forgot to add the generated configure script

**Comment by dnakamura** [comment]
@keithc-ca  should be all fixed now

File: closed/OpenJ9.gmk [hunk 3]
```diff
@@ -163,30 +164,32 @@ OPENJ9_STAGED_FILES += $(JDK_OUTPUTDIR)/lib/classlib.properties
...
 
 $(eval $(call SetupCopyFiles,OPENJ9_STAGED_FILES, \
 	DEST := $(JDK_OUTPUTDIR)/lib, \
-	SRC := $(OUTPUT_ROOT)/vm, \
+	SRC := $(OPENJ9_VM_BUILD_DIR), \
 	FILES := \
 		J9TraceFormat.dat \
 		OMRTraceFormat.dat \
 	))
-
```
Comment (keithc-ca) [inline closed/OpenJ9.gmk]: Please restore this blank line that marks a logical separation.

**Comment by dnakamura** [comment]
@keithc-ca fixed

**Comment by keithc-ca** [comment]
Jenkins compile xlinuxcm jdk8

**Comment by keithc-ca** [comment]
It seems something assumes the boot JDK is Java 8:
```
00:12:46.256  javac: invalid source release: 8
```

**Comment by keithc-ca** [comment]
Jenkins compile xlinuxcm jdk8

**Pull request merged**